components/Tabs/Tabs.js

```diff
@@ -1,60 +1,67 @@
 class TabLink {
-  constructor(tabElement){
+  constructor(tabElement){ 
     // assign this.tabElement to the tabElement DOM reference
     // this.tabElement;
-    
+    this.tabElement = tabElement;
     // Get the `data-tab` value from this.tabElement and store it here
     // this.tabData = ; 
-    
+    this.tabData = document.querySelector(`.tab[data-tab='${tabElement.dataset.tab}']`);
     // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
-    /*
+    
     // Check to see if this.tabData is equal to 'all'
-    if(){
+    this.all = document.querySelector(`.tab[data-tab='all']`);
+    if(this.all === this.tabData){
       // If `all` is true, select all cards regardless of their data attribute values
       // this.cards = ;
+      this.cards = document.querySelectorAll('.card');
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
       // this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab='${tabElement.dataset.tab}']`);
     }
-    /*
+    
 
      // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class. 
     // this.cards = Array.from(this.cards).map();
-
+    this.cards = Array.from(this.cards).map(e => new TabCard(e));
```

Comment:
these can be whatever name you want but you want to give better meaning to the naming you use so later on if you look back you know what 'e' is. I would have called this 'card' and if your iterating thru some tabs then maybe use 'tab'. You want to get in the habit of giving meaningful names.